referenceframe/worldstate.go

```diff
@@ -106,3 +107,71 @@ func (ws *WorldState) ObstaclesInWorldFrame(fs FrameSystem, inputs map[string][]
 	}
 	return transformGeometriesToWorldFrame(ws.Obstacles)
 }
+
+// BufferedWorldstate adds a buffer to all geometries, where buffer is in mm
+func BufferedWorldstate(ws *WorldState, buffer float64) (*WorldState, error) {
+	// TODO: add unit test for this
+	// TODO: need to handle tranforms as well
+	// TODO: this could be refactored
+	var gif []*GeometriesInFrame
+	for _, geosInFrame := range ws.Obstacles {
+		geoms := geosInFrame.Geometries()
+		parent := geosInFrame.Parent()
+		obstacles := []spatial.Geometry{}
+		for _, geo := range geoms {
+			cfg, err := spatial.NewGeometryConfig(geo)
```

Comment:
If you've already got the geometry in a `cfg`, you can pull the x/y/z/r/l values from the geometry config directly. No need to add a whole new method to geometries.